src/pip/_internal/commands/show.py

```diff
@@ -60,6 +60,9 @@ def search_packages_info(query):
         installed[canonicalize_name(p.project_name)] = p
 
     query_names = [canonicalize_name(name) for name in query]
+    missing = [pkg for pkg in query_names if pkg not in installed]
```

Comment:
It's cosmetic, but I would try to output the non-canonicalized names in the warning below by doing e.g. `[name for name, pkg in zip(query, query_names) if pkg not in installed]`.

Reply:
Good thinking that works well.

Reply:
Do you mind updating the test for it as well? Don't want the work to go to waste.

Reply:
Done in both tests _by changing requested name(s) to ones that will be "canonicalized"_ - with a comment.

Reply:
Thanks!